Support .NET Core 3.0 as the default

rodrmoya opened the pull request.

Commit:
[DotNetCore] Fix .NET Core 3.0 tests when only 3.0 is installed

Commit:
[PackageManagement] Fix .NET Core tests with only 3.0 installed

Commit:
[Debugger] Bump VS Code debugger protocol

Commit:
[Tests] Target NetCoreApp 3.0 for .NET Core console project

Commit:
[Tests] Set PATH to standard dotnet location to avoid other SDKs being used

Commit:
[DotNetCore] Add support for checking runtime in DotNetCoreSdkInstalledCondition

Commit:
[DotNetCore] Allow access to internals for MonoDevelop.AzureFunctions

Commit:
[Tests] Redirect MSBuild error output also

Therzok commented on main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreSdkInstalledCondition.cs:
```diff
@@ -66,6 +68,26 @@ public override bool Evaluate (NodeElement conditionNode)
+			DotNetCoreVersion.TryParse (requiredRuntimeVersion, out requiredDotNetCoreVersion);
+
+			if (requiredDotNetCoreVersion == null)
+				return versions.Any (version => version.ToString ().StartsWith (requiredRuntimeVersion, StringComparison.OrdinalIgnoreCase));
```
This looks wrong. Should we choose the latest non-preview?

Therzok commented on main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreSdkInstalledCondition.cs:
```diff
@@ -66,6 +68,26 @@ public override bool Evaluate (NodeElement conditionNode)
+			if (requiredDotNetCoreVersion == null)
+				return versions.Any (version => version.ToString ().StartsWith (requiredRuntimeVersion, StringComparison.OrdinalIgnoreCase));
+
+			return versions.Any (version => version.Major == requiredDotNetCoreVersion.Major && version.Minor == requiredDotNetCoreVersion.Minor);
```
```
requiredDotNetCoreVersion = 2.3.1
installedVersion1 = 2.3.0
installedVersion2 = 2.3.2
```

This will end up choosing the wrong version.

nosami replied:
Shouldn't it be first version >= required?

mrward replied:
Right now we do not use 3 digits for the requiredDotNetCoreVersion. Only use the first two. This is used for templates.

Therzok reviewed (dismissed):
Requesting changes on the method to choose the right version

nosami committed:
Remove FSharpSymbol Equals overload

nosami committed:
Test fix attempt

nosami committed:
Revert the .net core debugger bump

nosami committed:
bump

Therzok commented on .gitmodules:
```diff
@@ -1,6 +1,6 @@
 [submodule "main/external/debugger-libs"]
 	path = main/external/debugger-libs
-	url = git://github.com/mono/debugger-libs.git
+	url = https://github.com/nosami/debugger-libs.git
```
Submodule got retargeted by mistske

nosami replied:
It's not a mistake. I can't push to mono/debugger-libs and we need to make the MD/md-addins/debugger-libs merges at the same time

mrward committed:
[Core] Support getting FrameworkReferences for the project

Use ResolveFrameworkReferences MSBuild target to find framework
references for .NET Core 3.0 and .NET Standard 2.1 projects.

mrward committed:
[DotNetCore] Show FrameworkReferences in solution window

For .NET Core 3.0 and .NET Standard 2.1 projects a Frameworks folder
is now displayed under the Dependencies folder. This shows the
FrameworkReferences used by those projects. The SDK folder is hidden
for these projects since they do not have SDKs.
